invoices/invoiceregistry.go

```diff
@@ -99,21 +102,26 @@ func NewRegistry(cdb *channeldb.DB, finalCltvRejectDelta int32) *InvoiceRegistry
 		hodlSubscriptions:         make(map[channeldb.CircuitKey]map[chan<- interface{}]struct{}),
 		hodlReverseSubscriptions:  make(map[chan<- interface{}]map[channeldb.CircuitKey]struct{}),
 		finalCltvRejectDelta:      finalCltvRejectDelta,
+		expiryWatcher:             expiryWatcher,
 		quit:                      make(chan struct{}),
 	}
 }
 
 // Start starts the registry and all goroutines it needs to carry out its task.
 func (i *InvoiceRegistry) Start() error {
-	i.wg.Add(1)
+	i.expiryWatcher.Start(i.CancelInvoice)
+	i.expiryWatcher.PrefetchInvoices(i.cdb)
 
+	i.wg.Add(1)
```

Comment:
If this is adding the watcher to the wait group, it is probably better to create a new wait group inside the watcher and wait for it on `Stop`

Reply:
Maybe I misunderstand your comment but, `InvoiceExpiryWatcher` has its own wait group, and waits on it in `InvoiceExpiryWarcher.Stop()`.

Reply:
Ok, this `wg.Add` is just a move, but Github showed it a bit strange.